src/Lumper.UI/ViewModels/VtfBrowser/VtfBrowserViewModel.cs

```diff
@@ -0,0 +1,125 @@
+        {
+            get => _image;
+            set => this.RaiseAndSetIfChanged(ref _image, value);
+        }
+
+        public string Name
+        {
+            get => _name;
+            set => this.RaiseAndSetIfChanged(ref _name, value);
+        }
+    }
+
+    private double _dimensions = 128;
+
+    public double Dimensions
+    {
+        get => _dimensions;
+        set
+        {
+            this.RaiseAndSetIfChanged(ref _dimensions, value);
+            this.RaisePropertyChanged(nameof(MaxNameWidth));
+        }
+    }
+
+    public uint MaxNameWidth => (uint)_dimensions - 8;
+
+    private bool _showCubemaps;
+
+    public bool ShowCubemaps
+    {
+        get => _showCubemaps;
+        set
+        {
+            this.RaiseAndSetIfChanged(ref _showCubemaps, value);
+            this.RaisePropertyChanged(nameof(TextureBrowserItems));
+            this.RaisePropertyChanged(nameof(TexturesCount));
+        }
+    }
+
+    [GeneratedRegex(@"^((c-?\d+_-?\d+_-?\d+)|cubemapdefault)[\.hdr]{0,}\.vtf$")]
+    private static partial Regex _rgxCubemap();
+
+    private static ObservableCollection<Vtf> _textureBrowserItems =
+        new ObservableCollection<Vtf>();
+
+    public ObservableCollection<Vtf> TextureBrowserItems
+    {
+        get
+        {
+            var isGlobPattern = TextureSearch.Contains('*') || TextureSearch.Contains('?');
+            var matcher = isGlobPattern
+                ? new GlobMatcher(TextureSearch, false, true)
+                : new GlobMatcher($"*{TextureSearch}*", false, true);
+
+            var localMatcher = matcher;
+
+            return new ObservableCollection<Vtf>(
+                _textureBrowserItems.Where(t => // delegate allocation but other way is ugly lol
+                {
+                    if (!_showCubemaps && _rgxCubemap().IsMatch(t.Name))
+                    {
+                        return false;
+                    }
+
+                    return string.IsNullOrWhiteSpace(TextureSearch)
+                           || localMatcher.Match(t.Name).Result;
+                }));
+        }
+    }
+
+    private string _textureSearch = "";
+
+    public string TextureSearch
+    {
+        get => _textureSearch;
+        set
+        {
+            this.RaiseAndSetIfChanged(ref _textureSearch, value);
+            this.RaisePropertyChanged(nameof(TextureBrowserItems));
+            this.RaisePropertyChanged(nameof(TexturesCount));
+        }
+    }
+
+    public string TexturesCount => $"{TextureBrowserItems.Count} Items";
+
+    public static void AddTexture(PakFileEntry entry, string name)
+    {
+        //duplicate code from PakFileEntryVtfViewModel::Open
+        using var mem = new MemoryStream();
+        entry.DataStream.CopyTo(mem);
+        var vtfFileData = new VtfFileData(mem.ToArray());
+
+        _textureBrowserItems.Add(new Vtf
+        {
+            Name = name,
+            Image = vtfFileData.GetImage(0, 0, 0, 0)
+        });
+    }
```

Comment:
I don't like this, it get called in the constructor and creates *all* the images even if the user never opens the viewer .. it also does the same as the other PakFile thing (as your comment says) .. haven't decided how we should do this, but its either using the same `Image<Rgba32>` as the other class (so using the `Vtf` class in both or keeping a reference of the PakFile class here and using its image) or at least doing some of the work here in a getter so it doesn't get called right away .. because you need all images for the view, it will still use a lot of memory for this and we should probably clear some of it but I guess we can improve the viewers performance later .. the VTFs have a thumbnail I was playing around with but its very low resolution

Reply:
yeah we should discuss this on discord